src/main/java/org/broadinstitute/barclay/help/BashTabCompletionDoclet.java

```diff
@@ -0,0 +1,533 @@
+package org.broadinstitute.barclay.help;
+
+import com.sun.javadoc.RootDoc;
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * For testing of help documentation generation.
+ */
+public class BashTabCompletionDoclet extends HelpDoclet {
+
+    // Barclay BashTabCompletionDoclet custom Command-line Arguments:
+
+    final private static String CALLER_SCRIPT_NAME = "-caller-script-name";
+
+    final private static String CALLER_SCRIPT_PREFIX_LEGAL_ARGS          = "-caller-pre-legal-args";
+    final private static String CALLER_SCRIPT_PREFIX_ARG_VALUE_TYPES     = "-caller-pre-arg-val-types";
+    final private static String CALLER_SCRIPT_PREFIX_MUTEX_ARGS          = "-caller-pre-mutex-args";
+    final private static String CALLER_SCRIPT_PREFIX_ALIAS_ARGS          = "-caller-pre-alias-args";
+    final private static String CALLER_SCRIPT_PREFIX_ARG_MIN_OCCURRENCES = "-caller-pre-arg-min-occurs";
+    final private static String CALLER_SCRIPT_PREFIX_ARG_MAX_OCCURRENCES = "-caller-pre-arg-max-occurs";
+
+    final private static String CALLER_SCRIPT_POSTFIX_LEGAL_ARGS          = "-caller-post-legal-args";
+    final private static String CALLER_SCRIPT_POSTFIX_ARG_VALUE_TYPES     = "-caller-post-arg-val-types";
+    final private static String CALLER_SCRIPT_POSTFIX_MUTEX_ARGS          = "-caller-post-mutex-args";
+    final private static String CALLER_SCRIPT_POSTFIX_ALIAS_ARGS          = "-caller-post-alias-args";
+    final private static String CALLER_SCRIPT_POSTFIX_ARG_MIN_OCCURRENCES = "-caller-post-arg-min-occurs";
+    final private static String CALLER_SCRIPT_POSTFIX_ARG_MAX_OCCURRENCES = "-caller-post-arg-max-occurs";
+
+    // =============================================
+
+    // Variables that are set on the command line when running javadoc:
```

Comment:
All the command-line arguments are optional - I did that so the project could be run and instantly generate documentation without many setup woes.  I agree that the name probably should be provided, but it has a default name that it generates for the script which is somewhat descriptive ("launchScript-completion.sh").